subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -298,6 +307,29 @@ static void cleanup_connection(void)
 	}
 }
 
+static int string2hex(const char *string, const size_t len, char *hex)
```

Comment:
Zephyr already contains some of these functions as in `sys/util.h`. For this one, I think `hex2bin` does the job. Can you check?

Reply:
@sylvioalves I think you are right, didn't notice that `hex2bin` actually outputs `uint8_t` instead of binary as I previously thought, thanks for pointing that out.

Reply:
I modified it to use the `hex2bin` now, but I'm not sure if it works (unsure about the `strlen()` part), have to test it when I'm back to work